notebooks/examples/python/BritishNationalGrid.py

```diff
@@ -0,0 +1,370 @@
+    .withColumn("calculated_length", mos.st_length(col("geometry")))
+    # Note: The unit of measure of the area and length depends on the CRS used.
+    # For British National Grid locations it will be square meters and meters
+    .select("geometry", "calculated_area", "calculated_length")
+)
+
+# COMMAND ----------
+
+# MAGIC %md
+# MAGIC ## Read points data
+
+# COMMAND ----------
+
+# MAGIC %md
+# MAGIC We will load the Unique Property Reference Numbers (UPRNs) data to represent point data. </br>
+# MAGIC In order to setup the data please run the notebook available at "../../data/DownloadUPRNsData". </br>
+# MAGIC DownloadUPRNsData notebook will make sure we have UPRN table with point data available in our environment.
+# MAGIC We already loaded some shapes representing polygons that correspond to London postcodes. </br>
+
+# COMMAND ----------
+
+uprns_table = spark.table("uprns_table")
+display(uprns_table)
+
+# COMMAND ----------
+
+# MAGIC %md
+# MAGIC The UPRNs table contains Unique Property Reference Numbers and positions provided in EPSG:27700 and EPSG:4326. <br>
+# MAGIC Since we are operating in EPSG:27700 and using BNG as our indexing system, we will use the location data provided via Northings and Eastings coordinates.
+
+# COMMAND ----------
+
+uprns_table = (
+  uprns_table
+    .withColumn("uprn_point", mos.st_point(col("X_COORDINATE"), col("Y_COORDINATE")))
+    # we are using WKT here for simpler displaying, use WKB for faster query run time
+    .withColumn("uprn_point", mos.st_aswkt("uprn_point")) 
+    .where(mos.st_hasvalidcoordinates("uprn_point", lit('EPSG:27700'), lit('reprojected_bounds')))
+    .where(mos.st_isvalid(col("uprn_point")))
+    .drop("LATITUDE", "LONGITUDE")
+)
+display(uprns_table)
+
+# COMMAND ----------
+
+# MAGIC %md
+# MAGIC Next step is optional. Howerver, since we are constructing POINT geometries and ensuring they are valid it is prudent to write out the validated dataset. <br>
+# MAGIC That way we are making sure validation is performed only once ate ingestion and not each time spark runs the queries (due to spark lazy evaluation). 
+
+# COMMAND ----------
+
+uprns_table.write.format("delta").mode("overwrite").saveAsTable("uprns_bng_table")
+
+# COMMAND ----------
+
+uprns_table = spark.read.table("uprns_bng_table")
+
+# COMMAND ----------
+
+uprns_table.count()
+
+# COMMAND ----------
+
+# MAGIC %md
+# MAGIC ## Spatial Joins
+
+# COMMAND ----------
+
+# MAGIC %md
+# MAGIC We can use Mosaic to perform spatial joins both with and without Mosaic indexing strategies. </br>
+# MAGIC Indexing is very important when handling very different geometries both in size and in shape (ie. number of vertices). </br>
+
+# COMMAND ----------
+
+# MAGIC %md
+# MAGIC ### Getting the optimal resolution
+
+# COMMAND ----------
+
+# MAGIC %md
+# MAGIC We can use Mosaic functionality to identify how to best index our data based on the data inside the specific dataframe. </br>
```

Comment:
index -> tessellate

Reply:
I will use here index/tesselate.
People tend to use word index in these contexts when I speak with them.
Dont want to make this too hard to read for someone who isnt an expert.